builtin/providers/openstack/resource_openstack_compute_instance_v2.go

```diff
@@ -659,3 +671,45 @@ func getFloatingIPs(networkingClient *gophercloud.ServiceClient) ([]floatingips.
 	}
 	return ips, nil
 }
+
+func getImageID(client *gophercloud.ServiceClient, d *schema.ResourceData) (string, error) {
+	imageId := d.Get("image_id").(string)
```

Comment:
Here's a more idiomatic way of doing this (the formatting's off, but the idea is there: prefer early returns):
```
func getImageID(client *gophercloud.ServiceClient, d *schema.ResourceData) (string, error) {
	imageId := d.Get("image_id").(string)
        if imageId != "" {
                return imageId, nil
        }

        imageCount := 0
	imageName := d.Get("image_name").(string)
	imageName != "" {
		pager := images.ListDetail(client, &images.ListOpts{
			Name: imageName,
		})
		pager.EachPage(func(page pagination.Page) (bool, error) {
			imageList, err := images.ExtractImages(page)
			if err != nil {
				return false, err
			}

			for _, i := range imageList {
				if i.Name == imageName {
					imageCount++
					imageId = i.ID
				}
			}
			return true, nil
		})

		switch imageCount {
		case 0:
			return "", fmt.Errorf("Unable to find image: %s", imageName)
		case 1:
			return imageId, nil
		default:
			return "", fmt.Errorf("Found %d images matching %s", imageCount, imageName)
		}
	}
	return "", fmt.Errorf("Neither an image ID nor an image name were able to be determined.")
}
```